src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/ApplicationService.kt

```diff
@@ -778,4 +781,26 @@ class ApplicationService(
 
   fun getOfflineApplicationsForCrn(crn: String, serviceName: ServiceName) =
     offlineApplicationRepository.findAllByServiceAndCrn(serviceName.value, crn)
+
+  fun getApplicationTimeline(applicationId: UUID): List<DomainEventSummary> {
+    val domainEvents = domainEventService.getAllDomainEventsForApplication(applicationId)
+    val informationRequests = getAllInformationRequestEventsForApplication(applicationId)
+    return domainEvents + informationRequests
+  }
+
+  fun getAllInformationRequestEventsForApplication(applicationId: UUID): List<DomainEventSummary> {
+    val assessments = applicationRepository.findAllAssessmentsById(applicationId)
+
+    var allClarifications = mutableListOf<AssessmentClarificationNoteEntity>()
+    assessments.map {
+      allClarifications.addAll(it.clarificationNotes)
+    }
+    return allClarifications.map {
+      DomainEventSummary(
```

Comment:
This doesn't feel _quite_ right to me - this isn't really a Domain event as such, so I don't think it's right to marshal it into a `DomainEventSummary`. I reckon it might be better here to use a transformer in the service method itself and have two functions - one to convert a Domain event to the API object, and one to convert the clarification note into the API object.